skills/cryptocompare/api.py

```diff
@@ -0,0 +1,215 @@
+    headers = {
+        "Accept": "application/json",
+        "Authorization": f"Bearer {CRYPTO_COMPARE_API_KEY}"
+    }
+    params = {
+        "fsym": from_symbol.upper()
+    }
+    response = requests.get(url, params=params, headers=headers)
+    if response.status_code != 200:
+        return {"error": f"API returned status code {response.status_code}"}
+    return response.json()
+
+def fetch_top_market_cap(limit: int = 10, to_symbol: str = "USD") -> dict:
+    """
+    Fetch top cryptocurrencies by market cap.
+    """
+    url = f"{CRYPTO_COMPARE_BASE_URL}/data/top/mktcapfull"
+    headers = {
+        "Accept": "application/json",
+        "Authorization": f"Bearer {CRYPTO_COMPARE_API_KEY}"
+    }
+    params = {
+        "limit": limit,
+        "tsym": to_symbol.upper()
+    }
+    response = requests.get(url, params=params, headers=headers)
+    if response.status_code != 200:
+        return {"error": f"API returned status code {response.status_code}"}
+    return response.json()
+
+def fetch_top_exchanges(from_symbol: str, to_symbol: str = "USD") -> dict:
+    """
+    Fetch top exchanges for a cryptocurrency pair.
+    """
+    url = f"{CRYPTO_COMPARE_BASE_URL}/data/top/exchanges"
+    headers = {
+        "Accept": "application/json",
+        "Authorization": f"Bearer {CRYPTO_COMPARE_API_KEY}"
+    }
+    params = {
+        "fsym": from_symbol.upper(),
+        "tsym": to_symbol.upper()
+    }
+    response = requests.get(url, params=params, headers=headers)
+    if response.status_code != 200:
+        return {"error": f"API returned status code {response.status_code}"}
+    return response.json()
+
+def fetch_top_volume(limit: int = 10, to_symbol: str = "USD") -> dict:
+    """
+    Fetch top cryptocurrencies by total volume.
+    """
+    url = f"{CRYPTO_COMPARE_BASE_URL}/data/top/totalvolfull"
+    headers = {
+        "Accept": "application/json",
+        "Authorization": f"Bearer {CRYPTO_COMPARE_API_KEY}"
+    }
+    params = {
+        "limit": limit,
+        "tsym": to_symbol.upper()
+    }
+    response = requests.get(url, params=params, headers=headers)
+    if response.status_code != 200:
+        return {"error": f"API returned status code {response.status_code}"}
+    return response.json()
+
+def fetch_news(token: str, timestamp: int = None) -> dict:
+    """
+    Fetch news for a specific token and timestamp.
+    """
+    if timestamp is None:
+        timestamp = int(time.time())
+    url = f"{CRYPTO_COMPARE_BASE_URL}/data/v2/news/?lang=EN&lTs={timestamp}&categories={token}&sign=true"
+    headers = {
+        "Accept": "application/json",
+        "Authorization": f"Bearer {CRYPTO_COMPARE_API_KEY}"
+    }
+    response = requests.get(url, headers=headers)
+    if response.status_code != 200:
+        return {"error": f"API returned status code {response.status_code}"}
+    return response.json()
+
+# Prompt Strings
+FETCH_NEWS_PROMPT = """
```

Comment:
It is recommended to place the definitions of PROMPT and INPUT in the files of each skill for easier reference. This is just a suggestion and not mandatory.

Reply:
I found that these defined PROMPTs are not being used; the prompt for the skill is actually the description field in the skill class. Please use these prompts in the description, or simply delete them. In any case, it should be kept in one place; otherwise, it will lead to misunderstandings.